File: compiler/lib/implementation/print.js
// Copyright (c) 2011, the Dart project authors.  Please see the AUTHORS file
// for details. All rights reserved. Use of this source code is governed by a
// BSD-style license that can be found in the LICENSE file.

function native__Logger__printString(str) {
  if (isolate$workerPrint) {
    isolate$workerPrint(str);
  } else if (this.console) {
    this.console.log(str);
  } else if (this.write) {
    this.write(str);
    this.write('\n');
  }
}

File: pkg/analyzer_testing/lib/src/mock_packages/flutter/foundation/print.dart
// Copyright (c) 2025, the Dart project authors. Please see the AUTHORS file
// for details. All rights reserved. Use of this source code is governed by a
// BSD-style license that can be found in the LICENSE file.

import 'package:analyzer_testing/src/mock_packages/mock_library.dart';

final foundationPrintLibrary = MockLibraryUnit(
  'lib/src/foundation/print.dart',
  r'''
typedef DebugPrintCallback = void Function(String? message, {int? wrapWidth});

DebugPrintCallback debugPrint = (String? message, {int? wrapWidth}) {};
''',
);
